app/forms/incomeSources/cease/IncomeSourceEndDateForm.scala

```diff
@@ -38,9 +38,12 @@ class IncomeSourceEndDateForm @Inject()(val dateService: DateService) extends Cu
   val dateMustNotBeMissingDayAndYearField = "dateForm.error.dayAndYear.required"
   val dateMustNotBeMissingMonthAndYearField = "dateForm.error.monthAndYear.required"
   val dateMustNotBeInvalid = "error.invalid"
-  val dateMustNotBeInFuture = "dateForm.error.future"
-  val dateMustBeAfterBusinessStartDate = "dateFrom.error.beforeStartDate"
-  val dateMustNotBeBefore6April2015 = "incomeSources.cease.endDate.selfEmployment.error.beforeEarliestDate"
+
+  def dateMustNotBeInFuture(incomeSourceType: IncomeSourceType) = s"incomeSources.cease.endDate.${incomeSourceType.messagesCamel}.future"
```

Comment:
these methods more relevant to IncomeSourcceType rather than IncomeSourceEndDateForm.scala